libs/remote_storage/src/local_fs.rs

```diff
@@ -171,23 +169,22 @@ impl RemoteStorage for LocalFs {
             }
         }
 
-        // Note that PathBuf starts_with only considers full path segments, but
+        // Note that Utf8PathBuf starts_with only considers full path segments, but
         // object prefixes are arbitrary strings, so we need the strings for doing
         // starts_with later.
-        let prefix = full_path.to_string_lossy();
+        let prefix = full_path.to_string();
 
         let mut files = vec![];
         let mut directory_queue = vec![initial_dir.clone()];
         while let Some(cur_folder) = directory_queue.pop() {
             let mut entries = fs::read_dir(cur_folder.clone()).await?;
```

Comment:
Camino would have it's own https://docs.rs/camino/latest/camino/struct.Utf8Path.html#method.read_dir_utf8 which allows "internal translation" which I think would be enough here.